packages/ra-core/src/reducer/admin/resource/data.js

```diff
@@ -69,6 +71,42 @@ export default (previousState = initialState, { type, payload, meta }) => {
             .map(record => ({ ...record, ...payload.data }));
         return addRecords(updatedRecords, previousState);
     }
+    if (type === CRUD_DELETE_OPTIMISTIC) {
+        const newState = Object.keys(previousState).reduce((acc, id) => {
```

Comment:
You're doing an `omit` here, which can be written in one line in ES6:

```js
const { id, ...nextState } = previousState;
```

Reply:
mmm `id` is a variable, not a string key

Reply:
`const { [payload.id]: removed, ...newState } = previousState;` :tada:

Reply:
I :heart: es6

Reply:
💪 much better